src/main/takeDetection/takeDetection.ts

```diff
@@ -0,0 +1,104 @@
+import { Word, IndexRange } from 'sharedTypes';
+import {
+  InjectableTake,
+  InjectableTakeGroup,
+} from 'main/editDelete/injectTakeInfo';
+import getSimilarityScore from './sentenceSimilarity';
+
+export type Sentence = {
+  startIndex: number | null;
+  endIndex: number | null;
+  sentenceString: string;
+};
+
+export function findSentences(words: Word[]): Sentence[] {
+  let currentSentence: Sentence = {
+    startIndex: 0,
+    endIndex: null,
+    sentenceString: '',
+  };
+  const sentences: Sentence[] = [];
+  words.forEach((word, idx) => {
```

Comment:
Note - any refactor type stuff I'm going to say don't worry about for now, just get this merged - but I will suggest changes anyway and we can potentially return to them to clean this up after the demo is done - depends on if we deem it worthwhile or not.

Refactor suggestion - this loop would be much cleaner as a `reduce` - see `transcriptToCuts` for a very similar example. It can be tricky to get your head around if you haven't done functional / immutable programming before though, so probably leave it for now